app/src/userpages/components/StreamPage/Show/index.jsx

```diff
@@ -264,11 +241,67 @@ export class StreamShowView extends Component<Props, State> {
     }
 }
 
+function StreamLoader(props: Props) {
+    const isMounted = useIsMounted()
+    const propsRef = useRef(props)
+    propsRef.current = props
+    const { id: streamId } = props.match.params || {}
+
+    const initStream = useCallback(async (id: StreamId) => {
+        let { current: currentProps } = propsRef
+        await propsRef.current.getKeys()
+        if (!isMounted()) { return }
+        currentProps = propsRef.current
+        return Promise.all([
+            currentProps.getStream(id).then(() => {
+                if (!isMounted()) { return }
+                currentProps.openStream(id)
+                currentProps.initEditStream()
+            }),
+            currentProps.getMyStreamPermissions(id),
+        ])
+    }, [isMounted, propsRef])
+
+    const createStream = useCallback(async () => {
+        const { current: currentProps } = propsRef
+        const newStreamId = await currentProps.createStream()
+        if (!isMounted()) { return }
+        currentProps.history.replace(`${links.userpages.streamShow}/${newStreamId}`)
+        initStream(newStreamId)
+    }, [initStream, propsRef, isMounted])
+
+    const initStreamRef = useRef(initStream)
+    initStreamRef.current = initStream
+    const createStreamRef = useRef(createStream)
+    createStreamRef.current = createStream
```

Comment:
Idea here is to store the initial actions and assume they don't change?

Reply:
The `.current` gets set on every update, so it's the opposite. This is so we can use `initStream`/`createStream` in the below `useEffect`, but not pass them as deps to it, otherwise it would fire the effect any time the callbacks change. 



In this case, our callbacks will never actually change because none of their deps will change, but if they were to start changing identity it would also start executing the effect on a different schedule which may or may not be immediately obvious, so it seems safest to simply not rely on callback identity behaviour to begin with.